packages/react-jss/src/createHoc.js

```diff
@@ -121,23 +118,22 @@ export default function createHOC<
         this.unsubscribeId = themeListener.subscribe(this.context, this.setTheme)
       }
     }
-
+    // TODO: remove this lifecycle when migrating to new Context API
     componentWillReceiveProps(nextProps: OuterPropsType, nextContext: Context) {
       this.context = nextContext
-      const {dynamicSheet} = this.state
-      if (dynamicSheet) dynamicSheet.update(nextProps)
     }
 
-    componentWillUpdate(nextProps: OuterPropsType, nextState: State) {
-      if (isThemingEnabled && this.state.theme !== nextState.theme) {
-        const newState = this.createState(nextState, nextProps)
+    componentDidUpdate(prevProps: OuterPropsType, prevState: State) {
+      const {dynamicSheet} = this.state
+      if (dynamicSheet) dynamicSheet.update(this.props)
+
+      if (isThemingEnabled && this.state.theme !== prevState.theme) {
```

Comment:
I think this can be moved into a function called `handleThemeUpdate` which is the callback to the themeListener.subscribe

Reply:
Correct me if im wrong, but themeListener.subscribe only accept one callback. Do you mean to write another themeListener.subscribe? 

Another question: in the callback, how can I access to the previous theme (`prevState.theme`) to unmanage it?

Reply:
yes, themeListener.subscribe accepts one argument. I think the current one is called `setTheme`.  Make it so that the whole creating of the state and sheets when the theme changes are in there. The new theme is being passed as an argument.

The only thing which need to stay in componentDidUpdate is the `unmanage` of the old theme.

Reply:
Hi, so I was trying out your suggestion. Apparently, i can't call `createState` in `setTheme` because there's  some side effects from calling `createState` (I am not sure what precisely the side effect(s) is/are). Moving them causes some test fail.